tests/run-pass/threadleak_ignored.rs

```diff
@@ -0,0 +1,25 @@
+// ignore-windows: Concurrency on Windows is not supported yet.
+// compile-flags: -Zmiri-ignore-leaks
+
+//! Test that leaking threads works, and that their destructors are not executed.
+
+use std::cell::RefCell;
+
+struct LoudDrop(i32);
+impl Drop for LoudDrop {
+    fn drop(&mut self) {
+        eprintln!("Dropping {}", self.0);
+    }
+}
+
+thread_local! {
+    static X: RefCell<Option<LoudDrop>> = RefCell::new(None);
+}
+
+fn main() {
+    X.with(|x| *x.borrow_mut() = Some(LoudDrop(0)));
+    
+    let _detached = std::thread::spawn(|| {
+        loop { X.with(|x| *x.borrow_mut() = Some(LoudDrop(1))); }
```

Comment:
Now it depends on the scheduler again I think? If the thread runs we'll get lots of `Dropping 1` messages, one per time that we overwrite `X`

Reply:
Okay I implemented that.